static/app/components/core/input/inputGroup.stories.tsx

```diff
@@ -0,0 +1,107 @@
+
+// eslint-disable-next-line import/no-webpack-loader-syntax
+import types from '!!type-loader!sentry/components/core/input/inputGroup';
+
+export default storyBook('InputGroup', (story, APIReference) => {
+  APIReference(types.InputGroup);
+
+  story('Default', () => {
+    return (
+      <Fragment>
+        <p>
+          The <JSXNode name="Input" /> component comes in different sizes:
+        </p>
+        <InputGroup>
+          <InputGroup.LeadingItems disablePointerEvents>
+            <IconSearch />
+          </InputGroup.LeadingItems>
+          <InputGroup.Input placeholder="Search" />
+        </InputGroup>
+        <Input placeholder="hodl" />
+      </Fragment>
+    );
+  });
+
+  story('Non-interactive items', () => {
+    return (
+      <Fragment>
+        <p>
+          If we put items in the leading or trailing items that are not interactive, we
+          should set <code>disablePointerEvents</code> so that mouse clicks will fall
+          through to the <code>Input</code> underneath and trigger a focus event.
+        </p>
+        <Grid>
+          <InputGroup>
+            <InputGroup.LeadingItems disablePointerEvents>
+              <IconSearch />
+            </InputGroup.LeadingItems>
+            <InputGroup.Input placeholder="Search" />
+          </InputGroup>
+          <InputGroup>
+            <InputGroup.Input placeholder="Search" />
+            <InputGroup.TrailingItems disablePointerEvents>
+              <IconAttachment />
+            </InputGroup.TrailingItems>
+          </InputGroup>
+        </Grid>
+      </Fragment>
+    );
+  });
+
+  story('Matrix', () => {
+    return (
+      <Fragment>
+        <Matrix
+          render={({leadingItems, trailingItems}) => {
+            return (
+              <InputGroup>
+                {leadingItems ? (
+                  <InputGroup.LeadingItems disablePointerEvents>
+                    {leadingItems}
+                  </InputGroup.LeadingItems>
+                ) : null}
+                <InputGroup.Input placeholder="Search" />
+                {trailingItems ? (
+                  <InputGroup.TrailingItems>{trailingItems}</InputGroup.TrailingItems>
+                ) : null}
+              </InputGroup>
+            );
+          }}
+          selectedProps={['leadingItems', 'trailingItems']}
+          propMatrix={{
+            leadingItems: [null, <IconSearch key="leading-icon" size="sm" />],
+            trailingItems: [
+              null,
+              <IconAttachment key="trailing-icon" size="sm" />,
+              <Button
+                key="trailing-button"
+                borderless
+                icon={<IconSettings />}
+                size="zero"
+                aria-label="Toggle story representation"
+              />,
+            ],
+          }}
+        />
+      </Fragment>
+    );
+  });
+});
+
+const Grid = styled('div')`
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+  gap: ${space(2)};
+`;
```

Comment:
I think I've seen something similar before :) I have a PR that would change a bit how we write stories, would love your thoughts [here](https://github.com/getsentry/sentry/pull/85899), as we can probably streamline some of this better